src/Mvc/test/Mvc.IntegrationTests/ComplexRecordIntegrationTest.cs

```diff
@@ -0,0 +1,4285 @@
+            Assert.Equal("1", entry.RawValue);
+
+            entry = Assert.Single(modelState, e => e.Key == "Info.Key").Value;
+            Assert.Equal("location1", entry.AttemptedValue);
+            Assert.Equal("location1", entry.RawValue);
+
+            entry = Assert.Single(modelState, e => e.Key == "Info.Value.Zipcode").Value;
+            Assert.Equal("98052", entry.AttemptedValue);
+            Assert.Equal("98052", entry.RawValue);
+
+            entry = Assert.Single(modelState, e => e.Key == "Info.Value.GpsCoordinates").Value;
+            Assert.Equal("10,20", entry.AttemptedValue);
+            Assert.Equal("10,20", entry.RawValue);
+        }
+
+        private record Person5(string Name, IFormFile Photo);
+
+        // Regression test for #4802.
+        [Fact]
+        public async Task ReportsFailureToCollectionModelBinder()
+        {
+            // Arrange
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "parameter",
+                ParameterType = typeof(IList<Person5>),
+            };
+
+            var testContext = ModelBindingTestHelper.GetTestContext(request =>
+            {
+                SetFormFileBodyContent(request, "Hello world!", "[0].Photo");
+
+                // CollectionModelBinder binds an empty collection when value providers are all empty.
+                request.QueryString = new QueryString("?a=b");
+            });
+
+            var modelState = testContext.ModelState;
+            var metadata = GetMetadata(testContext, parameter);
+            var modelBinder = GetModelBinder(testContext, parameter, metadata);
+            var valueProvider = await CompositeValueProvider.CreateAsync(testContext);
+            var parameterBinder = ModelBindingTestHelper.GetParameterBinder(testContext);
+
+            // Act
+            var modelBindingResult = await parameterBinder.BindModelAsync(
+                testContext,
+                modelBinder,
+                valueProvider,
+                parameter,
+                metadata,
+                value: null);
+
+            // Assert
+            Assert.True(modelBindingResult.IsModelSet);
+
+            var model = Assert.IsType<List<Person5>>(modelBindingResult.Model);
+            var person = Assert.Single(model);
+            Assert.Null(person.Name);
+            Assert.NotNull(person.Photo);
+            using (var reader = new StreamReader(person.Photo.OpenReadStream()))
+            {
+                Assert.Equal("Hello world!", await reader.ReadToEndAsync());
+            }
+
+            Assert.True(modelState.IsValid);
+            var state = Assert.Single(modelState);
+            Assert.Equal("[0].Photo", state.Key);
+            Assert.Null(state.Value.AttemptedValue);
+            Assert.Empty(state.Value.Errors);
+            Assert.Null(state.Value.RawValue);
+        }
+
+        private record TestModel(TestInnerModel[] InnerModels);
```

Comment:
Why did this property / parameter lose its `Array.Empty<TestInnerModel>()` default❔

Reply:
I think `new TestInnerModel[] {}` or `new TestInnerModel[0]` should work